mmf/models/mmf_transformer.py

```diff
@@ -236,6 +238,13 @@ def preprocess_sample(self, sample_list: Dict[str, Any]) -> BaseTransformerInput
                 else:
                     image_modal = sample_list.image_feature_0
                 input_ids[modality.key] = self.image_encoder(image_modal)
+            else:
+                if modality.key in sample_list:
```

Comment:
This part will need to be changed to read the modality keys form a list as per changes in #568